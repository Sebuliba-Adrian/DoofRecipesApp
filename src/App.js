import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import Registration from './components/authentication/registration';
import LoginPage from './components/authentication/login';
import Categories from './components/categories/categories';
import Dashboard from './components/dashboard'

const App = () => (
  <BrowserRouter>
    <div>
      <Route path="(/|/login)" component={LoginPage} />
      <Route path="/registration" component={Registration} />
      <Route path="/dashboard" component={Dashboard} />
      {/* <Route path="/" component={Dashboard} /> */}
    </div>
  </BrowserRouter>
);

export default App;
export const APIUrl = 'http://localhost:5000/';
