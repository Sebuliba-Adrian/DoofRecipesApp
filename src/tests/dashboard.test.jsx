import React from 'react';
import axios from 'axios'; // v0.15.3
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Dashboard from '../components/dashboard';
import { APIUrl } from '../App';

global.localStorage = {
  setItem: () => {},
  removeItem: () => {},
  getItem: () => 't0k3n'
};
const props = {
  history: {
    replace: jest.fn()
  }
};

jest.useFakeTimers();

describe('Dashboard tests', () => {
  configure({ adapter: new Adapter() });
  const dashboard = mount(<Dashboard {...props} />);
  const instance = dashboard.instance();

  it('renders correctly', () => {
    const dashbrd = renderer.create(<Dashboard {...props} />).toJSON();

    console.log(dashbrd.children[0].children)
    expect(dashbrd.children.length).toBe(1)
    expect(dashbrd).toMatchSnapshot();
  });

  it('shows categories', () => {
    instance.viewCategories();
  });

  it('searches categories', () => {
    instance.search('');
  });
  it('can update page size', () => {
    instance.updatePageSize(3);
  });

  it('can do a GET network call for categories', () => {
    instance.request('getCategories', 'categories/1', 'GET', '');
  });

  it('can add categories', () => {
    instance.request('addCategories', 'categories/1', 'POST', '');
  });

  it('can update categories', () => {
    instance.request('updateCategory', 'categories/1', 'UPDATE', '');
  });

  it('can delete categories', () => {
    instance.request('deleteCategory', 'categories/1', 'DELETE', '');
  });

  it('can add recipe', () => {
    instance.request('addRecipe', 'categories', 'POST', '');
  });

  it('can update recipe', () => {
    instance.request('updateRecipe', 'categories/1/recipe/1', 'UPDATE', '');
  });

  it('can delete recipe', () => {
    instance.request('deleteRecipe', 'categories/1/recipe/1', 'DELETE', '');
  });

  it('can navigate properly', () => {
    instance.request('navigateRecipes', 'categories/1/recipe/1', 'GET', '');
  });

  it('can do a GET network call  for recipes', () => {
    instance.request('getRecipes', 'categories/1/recipes', 'GET', '');
  });

  it('can navigate through categories', () => {
    instance.onNavigateCategories('/categories?page=1');
  });
  it('can navigate through recipes', () => {
    instance.onNavigateRecipes('/categories/1/recipes?page=1');
  });
  it('can logout a user', () => {
    instance.request('logoutUser', '', 'GET', '');
  });

  it('can load from the server', () => {
    instance.loadFromServer(2);
  });

  it('shows recipes', () => {
    instance.viewRecipes({
      id: 1,
      name: 'Selected Category',
      description: 'This is the selected category',
      date_created: 'Wed, 20 Sep 2018 11:31:50 GMT',
      date_updated: 'Wed, 20 Sep 2018 11:31:50 GMT'
    });
  });
});
