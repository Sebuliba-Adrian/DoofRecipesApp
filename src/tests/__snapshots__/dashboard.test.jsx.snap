// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard tests renders correctly 1`] = `
<div
  className="container-fluid"
  styles={
    Object {
      "maxWidth": "500px",
    }
  }
>
  <div
    className="custom-navbar bg-cool-blue"
  >
    <div
      className=""
    >
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <a
            className="navbar-brand logo text-white"
            href="#/"
            onClick={[Function]}
          >
            <h3>
              <span
                className="fa fa-lg fa-cutlery mr-3 ml-3"
              />
              |   Doof! recipes
            </h3>
          </a>
        </div>
        <div
          className="col-md-4 text-center"
        >
          <input
            className="form-control mr-sm-2 search-box"
            onKeyDown={[Function]}
            placeholder="Search e.g c:the category name"
            type="text"
          />
        </div>
        <div
          className="col-md-4 text-right"
        >
          <div
            className="dropdown logout-dropdown"
          >
            <div
              className="fa fa-user-circle fa-lg text-white dropdown-toggle"
              data-toggle="dropdown"
              id="dropdownMenuLink"
            >
               
              <span
                className="logo"
              >
                <b>
                   
                  t0k3n
                   
                </b>
              </span>
            </div>
            <div
              aria-labelledby="dropdownMenuLink"
              className="dropdown-menu dropdown-cust"
            >
              <a
                className="dropdown-item"
                href="#/"
                id="logoutLink"
                onClick={[Function]}
              >
                Logout
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-4 categories"
    >
      <div>
        <div
          className="container"
        >
          <div
            className="card mt-3 mb-3"
          >
            <div
              className="card-block"
            >
              <div
                className="d-flex w-100 justify-content-between mb-2"
              >
                <p />
                <button
                  className="btn btn-sm btn-primary col-xs-12"
                  data-target="#addCategoryModal"
                  data-toggle="modal"
                >
                  <span
                    className="fa fa-plus pull-left"
                  />
                   Add category
                </button>
              </div>
              <nav
                aria-label="..."
              >
                <ul
                  className="pagination pagination-lg"
                />
              </nav>
            </div>
          </div>
        </div>
        <div
          className="modal fade"
          id="addCategoryModal"
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog"
            role="document"
          >
            <div
              className="modal-content"
            >
              <div
                className="modal-header"
              >
                <h5
                  className="modal-title"
                  id="exampleModalLabel"
                >
                  Create a category
                </h5>
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="modal"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    ×
                  </span>
                </button>
              </div>
              <div
                className="modal-body"
              >
                <label
                  htmlFor="name"
                >
                  Name
                </label>
                <div
                  className="input-group"
                  onChange={[Function]}
                >
                  <input
                    className="form-control"
                    id="name"
                    name="name"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <br />
                <label
                  htmlFor="description"
                >
                  Description
                </label>
                <div
                  className="input-group"
                >
                  <textarea
                    className="form-control"
                    id="description"
                    maxLength="140"
                    name="description"
                    onChange={[Function]}
                    rows="7"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="modal-footer"
              >
                <button
                  className="btn btn-secondary"
                  data-dismiss="modal"
                  onClick={undefined}
                  type="button"
                >
                  Cancel
                </button>
                <button
                  className="btn btn-primary"
                  data-dismiss="modal"
                  onClick={[Function]}
                  type="button"
                >
                  Submit
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-8 recipes"
    >
      <div>
        <div
          className="container items-container"
        >
          <div
            className="card mt-3 mb-3"
          >
            <div
              className="card-block"
            >
              <div
                className="d-flex w-100 justify-content-between mb-2"
              >
                <button
                  className="btn btn-sm btn-cool-blue col-xs-12"
                  data-target="#addRecipeModal"
                  data-toggle="modal"
                >
                  <span
                    className="fa fa-plus pull-left"
                  />
                   Add Recipe
                </button>
              </div>
              <nav
                aria-label="..."
              >
                <ul
                  className="pagination pagination-lg"
                />
              </nav>
              <div>
                <h6
                  className="card-text ml-4 pb-2 text-center text-cool-blue"
                >
                  Recipes in 
                </h6>
                <div
                  className="container"
                >
                  <div
                    className="row"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal fade"
          id="addRecipeModal"
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog"
            role="document"
          >
            <div
              className="modal-content"
            >
              <div
                className="modal-header"
              >
                <h5
                  className="modal-title"
                  id="exampleModalLabel"
                />
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="modal"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    ×
                  </span>
                </button>
              </div>
              <div
                className="modal-body"
              >
                <label
                  htmlFor="name"
                >
                  Name
                </label>
                <div
                  className="input-group"
                  onChange={[Function]}
                >
                  <input
                    className="form-control"
                    id="name"
                    name="name"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <br />
                <label
                  htmlFor="description"
                >
                  Description
                </label>
                <div
                  className="input-group"
                >
                  <textarea
                    className="form-control"
                    id="description"
                    maxLength="140"
                    name="description"
                    onChange={[Function]}
                    rows="7"
                    type="text"
                    value=""
                  />
                </div>
                <br />
              </div>
              <div
                className="modal-footer"
              >
                <button
                  className="btn btn-secondary"
                  data-dismiss="modal"
                  onClick={[Function]}
                  type="button"
                >
                  Cancel
                </button>
                <button
                  className="btn btn-primary"
                  data-dismiss="modal"
                  onClick={[Function]}
                  type="button"
                >
                  Submit
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    id="snackbar"
  />
</div>
`;
